src/foam/core/stdlib.js

```diff
@@ -568,9 +576,10 @@ foam.LIB({
     function getTime(d) { return ! d ? 0 : d.getTime ? d.getTime() : d ; },
     function equals(a, b) { return this.getTime(a) === this.getTime(b); },
     function compare(a, b) {
+      if ( ! this.isInstance(b) && ! foam.Number.isInstance(b) ) return -1;
```

Comment:
We re not. We assume that `b` will not be of a type "prior to" (in this case) `foam.Date` in `foam.typeOrder` (or equivalent `ordinal` markings). That means that, if `b` isn't a `Date` or something that can be interpreted as a `Date`, we wish to return "`a` comes before `b`": `-1`. The recommendation I make for calling these directly from outside of `foam.util.compare` only when `a` and `b` are (in this case) `Date`s is so that the behaviour is independent of these assumptions that come from type ordering.